3-platform/delivery/spec/src/main/java/com/_6core/platform/deliveryspec/rest/v1/DeliveryStatusManagementApi.java

```diff
@@ -0,0 +1,114 @@
+package com._6core.platform.deliveryspec.rest.v1;
+
+import com._6core.platform.deliveryspec.rest.v1.dto.OrderStatusResponse;
+import com._6core.platform.deliveryspec.rest.v1.dto.UpdatedOrderStatusResponse;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PatchMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import reactor.core.publisher.Mono;
+
+@Tag(
+    name = "Delivery Sataus Management Specification",
+    description = "REST reactive specification for menegment order delivery status")
+public interface DeliveryStatusManagementApi {
+
+  /**
+   * Exploring order status GET /delivery/{orderId}/status
+   *
+   * @param orderId (required)
+   * @return OK (status code 200)
+   */
+  @GetMapping(value = "/delivery/{orderId}/status", produces = "application/json")
```

Comment:
remember that URI should be a noun in plural form.
Structure like `first-nouns/{nounId}/second-nouns` is generally representing `relationships` between `first-nouns` and `second-nouns`. In therminology of RDBMS - one-to-many. many-to-many,...

So, we need to get rid of `status` in URI.
Consider using Query Param, e.g. `nouns/{nounId}/?include=status`. 
`Noun` to be replaced.
```suggestion
  @GetMapping(value = "/delivery/{orderId}/status", produces = "application/json")
```

Reply:
@sokimaaa, I'm considering not adding @QueryParam to endpoints for this Api. According to context requirements, I should get the order status by id and update it to the exact 'delivering' and 'delivered' statuses. So for these endpoints, we need only the order id. 
I'm not sure why I should use @QueryParam there.

Reply:
okay, lets discuss on friday